crates/host/src/wasmbus/mod.rs

```diff
@@ -436,25 +437,148 @@ async fn connect_nats(
     key: Option<Arc<KeyPair>>,
     require_tls: bool,
     request_timeout: Option<Duration>,
+    workload_identity_config: Option<WorkloadIdentityConfig>,
 ) -> anyhow::Result<async_nats::Client> {
-    let opts = async_nats::ConnectOptions::new().require_tls(require_tls);
-    let opts = match (jwt, key) {
-        (Some(jwt), Some(key)) => opts.jwt(jwt.to_string(), {
-            move |nonce| {
-                let key = key.clone();
-                async move { key.sign(&nonce).map_err(async_nats::AuthError::new) }
+    let opts =
+        match (jwt, key, workload_identity_config) {
+            (Some(jwt), Some(key), Some(wid_cfg)) => {
+                let jwt = jwt.to_owned();
+
+                let (tx, mut rx) = tokio::sync::mpsc::channel::<
+                    tokio::sync::oneshot::Sender<
+                        Result<spiffe::JwtSvid, spiffe::error::GrpcClientError>,
+                    >,
+                >(100);
+
```

Comment:
Would be great to find a way to handle this without the channels but yeesh the auth callback function is not agreeing with the spiffe workload API client, some deep-down trait isn't `Send` so it doesn't work. Messing with it locally

Reply:
I've incorporated your suggestions in 1d82ac78774a3a41a3325be552149e0badb96b16, reads a lot better now!